### ISSUE TITLE
[0.9.0]: fatal error: runtime: out of memory: Simple query crash the database

### ISSUE THREAD

**Issue by dtouzeau** [influxdb/influxdb#3109]
On a new server, Running a simple query trough the cli with a limit of 100 items crashes the database:
```
InfluxDB shell 0.9.0
> use proxydb
Using database proxydb
> SELECT * FROM access_log WHERE time >'2015-06-14 14:00:00' LIMIT 100
fatal error: runtime: out of memory

runtime stack:
runtime.SysMap(0xc22a680000, 0x100000, 0x7f2c589c6c00, 0xaca138)
        /root/.gvm/gos/go1.4.2/src/runtime/mem_linux.c:149 +0x98
runtime.MHeap_SysAlloc(0xacf820, 0x100000, 0x4368c2)
        /root/.gvm/gos/go1.4.2/src/runtime/malloc.c:284 +0x124
runtime.MHeap_Alloc(0xacf820, 0x1, 0x1000000000b, 0x415dae)
        /root/.gvm/gos/go1.4.2/src/runtime/mheap.c:240 +0x66
runtime.MCentral_CacheSpan(0xad6db8, 0x7f2c5735bc30)
        /root/.gvm/gos/go1.4.2/src/runtime/mcentral.c:85 +0x167
runtime.MCache_Refill(0x7f2c5b179000, 0xb, 0x7f2c5735bc30)
        /root/.gvm/gos/go1.4.2/src/runtime/mcache.c:90 +0xa0

goroutine 1 [running]:
runtime.switchtoM()
        /root/.gvm/gos/go1.4.2/src/runtime/asm_amd64.s:198 fp=0xc208091268 sp=0xc208091260
runtime.mallocgc(0xa0, 0x0, 0x3, 0x1)
        /root/.gvm/gos/go1.4.2/src/runtime/malloc.go:178 +0x849 fp=0xc208091318 sp=0xc208091268
runtime.rawstring(0x91, 0x0, 0x0, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.4.2/src/runtime/string.go:195 +0x93 fp=0xc208091348 sp=0xc208091318
runtime.slicebytetostring(0xc21b786120, 0x91, 0x120, 0x0, 0x0)
        /root/.gvm/gos/go1.4.2/src/runtime/string.go:69 +0x4b fp=0xc2080913d8 sp=0xc208091348
fmt.Sprintf(0x8b6c90, 0x8, 0xc2080915a0, 0x1, 0x1, 0x0, 0x0)
        /root/.gvm/gos/go1.4.2/src/fmt/print.go:204 +0x99 fp=0xc208091428 sp=0xc2080913d8
main.(*CommandLine).formatResults(0xc208074a50, 0xc21e460000, 0x14e55, 0x1680a, 0x0, 0x0, 0x883730, 0x1, 0x0, 0x0, ...)
        /root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influx/main.go:572 +0x107a fp=0xc208091818 sp=0xc208091428
main.(*CommandLine).writeColumns(0xc208074a50, 0xc2080c0930, 0x7f2c5b18ae70, 0xc20802e008)
        /root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influx/main.go:519 +0x177 fp=0xc2080919d0 sp=0xc208091818
main.(*CommandLine).FormatResponse(0xc208074a50, 0xc2080c0930, 0x7f2c5b18ae70, 0xc20802e008)
        /root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influx/main.go:481 +0xc2 fp=0xc208091a70 sp=0xc2080919d0
main.(*CommandLine).ExecuteQuery(0xc208074a50, 0xc20800acd0, 0x44, 0x0, 0x0)
        /root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influx/main.go:462 +0x218 fp=0xc208091b68 sp=0xc208091a70
main.(*CommandLine).ParseCommand(0xc208074a50, 0xc20800acd0, 0x44, 0xc20800acd0)
        /root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influx/main.go:220 +0xa23 fp=0xc208091d48 sp=0xc208091b68
main.main()
        /root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influx/main.go:169 +0xc0e fp=0xc208091f98 sp=0xc208091d48
runtime.main()
        /root/.gvm/gos/go1.4.2/src/runtime/proc.go:63 +0xf3 fp=0xc208091fe0 sp=0xc208091f98
runtime.goexit()
        /root/.gvm/gos/go1.4.2/src/runtime/asm_amd64.s:2232 +0x1 fp=0xc208091fe8 sp=0xc208091fe0

goroutine 5 [syscall, 1 minutes]:
os/signal.loop()
        /root/.gvm/gos/go1.4.2/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
        /root/.gvm/gos/go1.4.2/src/os/signal/signal_unix.go:27 +0x35

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
        /root/.gvm/gos/go1.4.2/src/runtime/asm_amd64.s:2232 +0x1
```

**Comment by dgnorton** [comment]
@dtouzeau looks like it was the CLI that paniced.  Did the database continue running?  Can you reproduce the panic?

**Comment by dtouzeau** [comment]
After this error I need to restart the influx service,  
Error is not generated only with limit=10, if I increase the limit to 100 I get the same error or the whole system crash (reboot) !!!. 
It seems that is only happens when database is over 1GB, 
if I remove the database and create a new fresh one the problem is not seen....until it reach 1GB data
I will try the same query with SLIMIT but it seems that will LIMIT, Influx should not crash

**Comment by beckettsean** [comment]
How much RAM is available to the process? How many points, approximately, are in the `access_log` measurement?

**Comment by dtouzeau** [comment]
points, some millions
2GB memory with only infludb runing

**Comment by dtouzeau** [comment]
updated, only 40992 lines

**Comment by pauldix** [comment]
@dgnorton SLIMIT won't do anything for his query because he doesn't have a `GROUP BY` with a tag. It's only one series. If this causes the server to run out of memory, it's a bug.

The `LIMIT 100` part of the query means that the engine should stop after reading 100 points, which I assume is very little in terms of memory usage.

**Comment by beckettsean** [comment]
@dtouzeau does this happen with 0.9.1?

**Comment by dtouzeau** [comment]
No it doesn't crash

**Comment by beckettsean** [comment]
resolved (or at least not reproducible) with 0.9.1, closing
